src/bun.js/api/server.zig

```diff
@@ -91,6 +91,58 @@ const SendfileContext = struct {
 const DateTime = bun.DateTime;
 const linux = std.os.linux;
 
+const BlobFileContentResult = struct {
+    data: [*c]const u8,
+
+    const log = Output.scoped(.BlobFileContentResult, false);
+    fn init(comptime fieldname: []const u8, js_obj: JSC.JSValue, global: *JSC.JSGlobalObject, exception: JSC.C.ExceptionRef) ?BlobFileContentResult {
+        if (JSC.WebCore.Body.Value.fromJS(global, js_obj)) |body| {
+            if (body == .Blob and body.Blob.store != null and body.Blob.store.?.data == .file) {
+                const pathlike = body.Blob.store.?.data.file.pathlike;
+                if (pathlike == .path) {
+                    const path = pathlike.path.slice();
+
+                    const path_z = bun.default_allocator.dupeZ(u8, path) catch unreachable;
+                    defer bun.default_allocator.destroy(path_z);
+                    if (std.os.system.access(path_z, std.os.F_OK) != 0) {
+                        JSC.throwInvalidArguments(std.fmt.comptimePrint("Unable to access {s} path", .{fieldname}), .{}, global, exception);
+                        return .{ .data = null };
+                    }
+                    const cwd = std.process.getCwdAlloc(bun.default_allocator) catch {
+                        JSC.throwInvalidArguments(std.fmt.comptimePrint("Unable to access {s} path", .{fieldname}), .{}, global, exception);
+                        return .{ .data = null };
+                    };
+                    defer bun.default_allocator.free(cwd);
+
+                    const fullpath = std.fs.path.resolve(bun.default_allocator, &.{ cwd, path }) catch {
```

Comment:
`std.fs.path.resolve` doesn't actually go through the filesystem so it doesn't do as intended

Reply:
instead, use `bun.getFdPath`

Reply:
ignore this, see above